internal/server/routes.go

```diff
@@ -395,16 +397,24 @@ func del[Req any, Res any](a *API, r *routeGroup, path string, handler HandlerFu
 }
 
 func readRequest(c *gin.Context, req interface{}) error {
-	if err := c.ShouldBindUri(req); err != nil {
-		return fmt.Errorf("%w: %s", internal.ErrBadRequest, err)
+	if len(c.Params) > 0 {
+		params := make(map[string][]string)
+		for _, v := range c.Params {
+			params[v.Key] = []string{v.Value}
+		}
+		if err := binding.Uri.BindUri(params, req); err != nil {
+			return fmt.Errorf("%w: %s", internal.ErrBadRequest, err)
+		}
 	}
 
-	if err := c.ShouldBindQuery(req); err != nil {
-		return fmt.Errorf("%w: %s", internal.ErrBadRequest, err)
+	if len(c.Request.URL.Query()) > 0 {
+		if err := binding.Query.Bind(c.Request, req); err != nil {
```

Comment:
By calling `binding` functions here directly it will allow us to replace this library with a much smaller implementation.

There's no functional change here. If you trace the cals of `c.ShouldBind` that I've replaced here you'll see they eventually call these functions (and do nothing else).